addons/editor-dark-mode/paint_icons.js

```diff
@@ -0,0 +1,70 @@
+import { textColor } from "../../libraries/common/cs/text-color.esm.js";
+
+const iconSelector = "[class*='labeled-icon-button_edit-field-icon_']";
+const dataUriRegex = new RegExp("^data:image/svg\\+xml;base64,([A-Za-z0-9+/=]*)$");
+
+export default async function ({ addon, console }) {
+  const recolorSvg = (svg) => {
+    return svg
+      .replace(/#855cd6/gi, addon.settings.get("highlightText"))
+      .replace(/#ffffff|#fff|white/gi, addon.settings.get("accent"))
+      .replace(/#575e75/gi, textColor(addon.settings.get("accent")));
+  };
+
+  const srcToSvg = async (src) => {
+    const match = dataUriRegex.exec(src);
+    if (match) return atob(match[1]);
+    return await (await fetch(src)).text();
+  };
+
+  const updateIcon = async (icon) => {
+    let oldSvg;
+    if (icon.saOriginalSvg) {
+      oldSvg = icon.saOriginalSvg;
+    } else {
+      oldSvg = await srcToSvg(icon.src);
+      icon.saOriginalSvg = oldSvg;
+    }
+    const newSvg = addon.self.disabled ? oldSvg : recolorSvg(oldSvg);
+    icon.src = `data:image/svg+xml;base64,${btoa(newSvg)}`;
+
+    // React sometimes changes the src of an existing icon instead of creating a new one
+    if (!icon.saOverriddenSetAttribute) {
+      icon.saOverriddenSetAttribute = true;
+      const oldSetAttribute = icon.setAttribute;
+      icon.setAttribute = function (name, value) {
+        if (name === "src") {
+          srcToSvg(value).then((svg) => {
+            icon.saOriginalSvg = svg;
+            updateIcon(icon);
+          });
+        }
+        return oldSetAttribute.call(this, name, value);
+      };
```

Comment:
Maybe the `oldSetAttribute.call` line can be moved above the "if" so that the order of operations is more clear? Remember setAttribute always returns undefined.